### PR TITLE
[MBL-17613][Parent] Implement Masquerading

### PR DESCRIPTION
**Pull request by tamaskozmer**
Test plan: Test all types of masquerading in the Parent app (qr code, site admin login, admin login)

refs: MBL-17613
affects: Parent
release note: none

## Checklist

- [x] Tested in dark mode
- [x] Tested in light mode

### CODE REVIEW

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Native Parent Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Review by kdeakinstructure — APPROVED** [review]
QA 👍

File: apps/parent/src/main/java/com/instructure/parentapp/features/splash/SplashViewModel.kt [hunk 1]
```diff
@@ -39,12 +41,15 @@ class SplashViewModel @Inject constructor(
...
     private val apiPrefs: ApiPrefs,
-    private val colorKeeper: ColorKeeper
+    private val colorKeeper: ColorKeeper,
+    savedStateHandle: SavedStateHandle
 ) : ViewModel() {
 
     private val _events = Channel<SplashAction>()
     val events = _events.receiveAsFlow()
 
+    private val qrMasqueradeId = savedStateHandle.get<Long>(Const.QR_CODE_MASQUERADE_ID) ?: 0L
```
Comment (hermannakos) [inline apps/parent/src/main/java/com/instructure/parentapp/features/splash/SplashViewModel.kt:51]: Could this be null instead? We're already checking for the magic 0. Or at least -1.

**Review by hermannakos — APPROVED** [review]

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Native Parent Install Page</p>

**Review by kristofnemere — APPROVED** [review]
QA+1

**Pull request merged**